File: src/minecraft/monnef/jaffas/food/block/BlockLeavesBaseJaffas.java
package monnef.jaffas.food.block;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.block.material.Material;
import net.minecraft.world.IBlockAccess;

public class BlockLeavesBaseJaffas extends BlockJaffas {
    public boolean graphicsLevel;

    protected BlockLeavesBaseJaffas(int par1, int texture, Material par2Material, boolean par3) {
        super(par1, texture, par2Material);
        this.graphicsLevel = par3;
    }

    @Override
    public boolean isOpaqueCube() {
        return false;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public boolean shouldSideBeRendered(IBlockAccess par1IBlockAccess, int par2, int par3, int par4, int par5) {
        int i1 = par1IBlockAccess.getBlockId(par2, par3, par4);
        return !this.graphicsLevel && i1 == this.blockID ? false : super.shouldSideBeRendered(par1IBlockAccess, par2, par3, par4, par5);
    }
}

File: src/minecraft/monnef/jaffas/food/block/BlockLeavesBaseJaffas.scala
/*
 * Jaffas and more!
 * author: monnef
 */

package monnef.jaffas.food.block

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import net.minecraft.block.material.Material
import net.minecraft.world.IBlockAccess
import monnef.jaffas.trees.common.IconDescriptorTrees

class BlockLeavesBaseJaffas(_id: Int, _texture: Int, _material: Material, var graphicsLevel: Boolean) extends BlockJaffas(_id, _texture, _material) with IconDescriptorTrees {

  override def isOpaqueCube: Boolean = false

  @SideOnly(Side.CLIENT) override def shouldSideBeRendered(par1IBlockAccess: IBlockAccess, par2: Int, par3: Int, par4: Int, par5: Int): Boolean = {
    val i1: Int = par1IBlockAccess.getBlockId(par2, par3, par4)
    if (!this.graphicsLevel && i1 == this.blockID) false
    else super.shouldSideBeRendered(par1IBlockAccess, par2, par3, par4, par5)
  }
}